axelor-base/src/main/java/com/axelor/apps/base/service/imports/ImportCityServiceImpl.java

```diff
@@ -153,4 +180,145 @@ private void deleteTempFiles(File configXmlFile, File dataCsvFile) {
       e.printStackTrace();
     }
   }
+
+  /**
+   * extracting file from the zip
+   *
+   * @param downloadFileName : zip fileName to download from internet
+   * @return
+   */
+  @Override
+  public void downloadZip(String downloadFileName) throws IOException {
+    File downloadFile;
+
+    File cityTextFile;
+    ZipFile zipFile;
+
+    URL url;
+
+    Path path =
+        Paths.get(
+            AppSettings.get()
+                .get(
+                    "file.upload.dir",
+                    "{java.io.tmpdir}"
+                        + File.separator
+                        + "axelor"
+                        + File.separator
+                        + "attachments"));
+
+    downloadFile = new File(path + File.separator + downloadFileName);
+
+    cityTextFile = new File(path + File.separator + "cityTextFile.txt");
+
+    url = new URL("http://download.geonames.org/export/zip/" + downloadFileName);
+
+    try {
+      downloadFile.createNewFile();
+    } catch (IOException e) {
+      Files.createParentDirs(downloadFile);
+      downloadFile.createNewFile();
+    }
+
+    /** checking whether geoname file already exists or not */
+    if (downloadFile.length() == Long.parseLong("0")) {
+      FileUtils.copyURLToFile(url, downloadFile);
+    }
+
+    zipFile = new ZipFile(path + File.separator + downloadFileName);
+
+    Enumeration<? extends ZipEntry> entries = zipFile.entries();
+
+    LOG.debug(path.toString());
```

Comment:
maybe we don't get a clear idea that what is printing in a log 
it's better to append some message suffix